tests/linear_model/test_glm.py

```diff
@@ -203,3 +219,121 @@ def test_logistic_predict_proba_shape():
     lr.fit(X, y)
     prob = lr.predict_proba(X)
     assert prob.shape == (100, 2)
+
+
+@pytest.mark.parametrize(
+    "reg_type,data_generator",
+    [
+        (LinearRegression, "single_chunk_regression"),
+        (LogisticRegression, "single_chunk_classification"),
+        (PoissonRegression, "single_chunk_count_classification"),
+    ],
+)
+def test_model_coef_dask_numpy(reg_type, data_generator, request):
+    """Tests that models return same coefficients and intercepts with array types"""
+    X, y = request.getfixturevalue(data_generator)
+    np_mod, da_mod = reg_type(fit_intercept=True), reg_type(fit_intercept=True)
+    da_mod.fit(X, y)
+    np_mod.fit(X.compute(), y.compute())
+    da_coef = np.hstack((da_mod.coef_, da_mod.intercept_))
+    np_coef = np.hstack((np_mod.coef_, np_mod.intercept_))
+
+    rel_error = LA.norm(da_coef - np_coef) / LA.norm(np_coef)
+    assert rel_error < 1e-8
+
+
+@pytest.mark.parametrize("solver", ["newton", "gradient_descent", "lbfgs"])
+@pytest.mark.parametrize("fit_intercept", [True, False])
+@pytest.mark.parametrize(
+    "reg_type, skl_reg_type, skl_params, data_generator",
+    [
+        (LinearRegression, slm.LinearRegression, {}, "medium_size_regression"),
+        (
+            LogisticRegression,
+            slm.LogisticRegression,
+            {"penalty": "none"},
+            "single_chunk_classification",
+        ),
+        (
+            PoissonRegression,
+            slm.PoissonRegressor,
+            {"alpha": 0},
+            "medium_size_counts",
+        ),
+    ],
+)
+def test_model_against_sklearn(
+    reg_type, skl_reg_type, skl_params, data_generator, fit_intercept, solver, request
+):
+    """
+    Tests accuracy of model predictions
+    """
+    X, y = request.getfixturevalue(data_generator)
+    dask_ml_model = reg_type(
+        fit_intercept=fit_intercept, solver=solver, penalty="l2", C=1e8, max_iter=500
+    )
+    skl_model = skl_reg_type(fit_intercept=fit_intercept, **skl_params)
+    # skl_model has to be fit with numpy data
+    dask_ml_model.fit(X, y)
+    skl_model.fit(X.compute(), y.compute())
+
+    est, truth = dask_ml_model.predict(X), skl_model.predict(X)
+    rel_error = LA.norm(est - truth) / LA.norm(truth)
+    assert rel_error < 1e-3
+
+
+@pytest.mark.parametrize("solver", ["newton", "gradient_descent", "lbfgs"])
+@pytest.mark.parametrize(
+    "reg_type, skl_reg_type, skl_params, dataset_function",
+    [
+        (LinearRegression, slm.LinearRegression, {}, make_regression),
+        (
+            LogisticRegression,
+            slm.LogisticRegression,
+            {"penalty": "none"},
+            make_classification,
+        ),
+    ],
+)
+def test_model_coef_against_sklearn(
```

Comment:
Maybe this test should be combined with `test_model_against_sklearn`?

Also, relative error tests are best suited for tests on the coefficients (and not so much for binary predictions).

Reply:
Great. I have combined the tests into a single parameterized test.